props/saber_BC_buttons.h

```diff
@@ -477,738 +1493,1269 @@ class SaberBCButtons : public PROP_INHERIT_PREFIX PropBase {
-          sound_library_.SayTheBatteryLevelIs();
-          sound_library_.SayDisabled();
-          return true;
-        }
-        sound_library_.SayTheBatteryLevelIs();
-        sound_library_.SayNumber(battery_monitor.battery(), SAY_DECIMAL);
-        sound_library_.SayVolts();
-        PVLOG_NORMAL << "Battery Voltage: " << battery_monitor.battery() << "\n";
-        speaking_ = true;
-        SaberBase::DoEffect(EFFECT_BATTERY_LEVEL, 0);
-      }
-      return true;
+  void ToggleSpamBlast() {
+    spam_blast_ = !spam_blast_;
+    PVLOG_NORMAL << (spam_blast_ ? "** Entering" : "** Exiting") << " Spam Blast Mode\n";
+    if (!hybrid_font.PlayPolyphonic(&SFX_mzoom)) {
+      spam_blast_ ? BeepEnterFeature() : BeepExitFeature();
+    }
+  }
 
-// Spoken Battery Level in percentage
-    case EVENTID(BUTTON_POWER, EVENT_THIRD_HELD, MODE_OFF):
-      if (!mode_volume_) {
-        if (battery_monitor.battery() < 0.5) {
-          sound_library_.SayTheBatteryLevelIs();
-          sound_library_.SayDisabled();
-          return true;
-        }
-        sound_library_.SayTheBatteryLevelIs();
-        sound_library_.SayNumber(battery_monitor.battery_percent(), SAY_WHOLE);
-        sound_library_.SayPercent();
-        PVLOG_NORMAL << "Battery Percentage: " <<battery_monitor.battery_percent() << "%\n";
-        speaking_ = true;
-        SaberBase::DoEffect(EFFECT_BATTERY_LEVEL, 0);
+  void ToggleBattleMode() {
+#ifndef BC_NO_BM
+  battle_mode_ = !battle_mode_;
+    PVLOG_NORMAL << (battle_mode_ ? "** Entering" : "** Exiting") << " Battle Mode\n";
+    if (battle_mode_) {
+      if (!hybrid_font.PlayPolyphonic(&SFX_bmbegin)) {
+        hybrid_font.DoEffect(EFFECT_FORCE, 0);
       }
-      return true;
-
-// On Demand Battery Level
-    case EVENTID(BUTTON_POWER, EVENT_SECOND_SAVED_CLICK_SHORT, MODE_OFF):
-      if (!mode_volume_) {
-        PVLOG_NORMAL << "Battery Voltage: " << battery_monitor.battery() << "\n";
-        PVLOG_NORMAL << "Battery Percentage: " <<battery_monitor.battery_percent() << "\n";
-        SaberBase::DoEffect(EFFECT_BATTERY_LEVEL, 0);
-      } else {
-        QuickMinVolume();
+    } else {
+      if (!hybrid_font.PlayPolyphonic(&SFX_bmend)) {
+        beeper.Beep(0.5, 3000);
       }
-      return true;
+    }
+#endif
+  }
 
-// Stab
-    case EVENTID(BUTTON_NONE, EVENT_THRUST, MODE_ON):
-      //Don't stab if in colorchange mode
-      if (SaberBase::GetColorChangeMode() != SaberBase::COLOR_CHANGE_MODE_NONE) return false;
-      if (SaberBase::IsOn()) { // prevents triggering during preon
-        SaberBase::SetClashStrength(2.0);
-        SaberBase::DoStab();
+#ifdef BC_DUAL_BLADES
+
+  static const BladeSet controlled_blades_;
+
+  bool isMainBladeOn() { return SaberBase::OnBlades()[BC_MAIN_BLADE]; }
+  bool isSecondBladeOn() { return SaberBase::OnBlades()[BC_SECOND_BLADE]; }
+
+  // Get blade above horizon line to apply effect to
+  BladeSet GetBladeAboveHorizon(bool return_opposite = false) {
```

Comment:
return_opposite is a makes a travesty of the name of this function.
Why not just have a GetBladeBelowHorizon() instead?

Reply:
This whole double-negative thing had me crazy and this worked so I left it. Agreed, return_opposite is not great.
Looking now, there's no reason we can't just do drag to the right blade directly in DualBladesLockupOrDrag().
Then GetBladeAboveHorizon() can be just what it says to handle clash, blast, and lockup.
Tested and works the majority of the time.
I think blade wiggle/vibrations messes with accel sometimes so the "wrong" blade shows impact effects, but it's mostly acceptable I think?
Edits done.